exchanges/kucoin/kucoin.go

```diff
@@ -0,0 +1,1864 @@
+}
+
+// GetDepositAddressV2 get all deposit addresses for the currency you intend to deposit
+func (ku *Kucoin) GetDepositAddressV2(ctx context.Context, ccy string) ([]DepositAddress, error) {
+	if ccy == "" {
+		return nil, currency.ErrCurrencyCodeEmpty
+	}
+	params := make(map[string]interface{})
+	params["currency"] = ccy
+	var resp []DepositAddress
+	return resp, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetDepositAddressV2, params, &resp)
+}
+
+// GetDepositAddressV1 get a deposit address for the currency you intend to deposit
+func (ku *Kucoin) GetDepositAddressV1(ctx context.Context, ccy, chain string) ([]DepositAddress, error) {
+	if ccy == "" {
+		return nil, currency.ErrCurrencyCodeEmpty
+	}
+	params := make(map[string]interface{})
+	params["currency"] = ccy
+	if chain != "" {
+		params["chain"] = chain
+	}
+	var resp []DepositAddress
+	return resp, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetDepositAddressV1, params, &resp)
+}
+
+// GetDepositList get deposit list items and sorted to show the latest first
+func (ku *Kucoin) GetDepositList(ctx context.Context, ccy, status string, startAt, endAt time.Time) ([]Deposit, error) {
+	params := make(map[string]interface{})
+	if ccy != "" {
+		params["currency"] = ccy
+	}
+	if status != "" {
+		params["status"] = status
+	}
+	if !startAt.IsZero() {
+		params["startAt"] = strconv.FormatInt(startAt.UnixMilli(), 10)
+	}
+	if !endAt.IsZero() {
+		params["endAt"] = strconv.FormatInt(endAt.UnixMilli(), 10)
+	}
+	resp := struct {
+		CurrentPage int64     `json:"currentPage"`
+		PageSize    int64     `json:"pageSize"`
+		TotalNum    int64     `json:"totalNum"`
+		TotalPage   int64     `json:"totalPage"`
+		Items       []Deposit `json:"items"`
+	}{}
+	return resp.Items, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetDepositList, params, &resp)
+}
+
+// GetHistoricalDepositList get historical deposit list items
+func (ku *Kucoin) GetHistoricalDepositList(ctx context.Context, ccy, status string, startAt, endAt time.Time) ([]HistoricalDepositWithdrawal, error) {
+	params := make(map[string]interface{})
+	if ccy != "" {
+		params["currency"] = ccy
+	}
+	if status != "" {
+		params["status"] = status
+	}
+	if !startAt.IsZero() {
+		params["startAt"] = strconv.FormatInt(startAt.UnixMilli(), 10)
+	}
+	if !endAt.IsZero() {
+		params["endAt"] = strconv.FormatInt(endAt.UnixMilli(), 10)
+	}
+	resp := struct {
+		CurrentPage int64                         `json:"currentPage"`
+		PageSize    int64                         `json:"pageSize"`
+		TotalNum    int64                         `json:"totalNum"`
+		TotalPage   int64                         `json:"totalPage"`
+		Items       []HistoricalDepositWithdrawal `json:"items"`
+	}{}
+	return resp.Items, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetHistoricalDepositList, params, &resp)
```

Comment:
As asked in the previous PR, why are you filtering API results? This makes a user less informed. Please return the full results in all responses